### PR TITLE
[Feature] Adding new settings page to the demo app instead of the entry view

### PR DESCRIPTION
**Pull request by JoshuaLai**
## Purpose
Adding setting to demo app to be more convenient 

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Launch the demo app and check the new settings 

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by jimchou-dev — APPROVED** [review]
Should we have the Settings button at top right to match Android?
image

Or maybe to be simple have the Settings button below `Start Experience` is a bit more logical to me.
Tested, LGTM

**Review by AmyL219 — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Views/UIKitDemoViewController.swift [hunk 1]
```diff
@@ -262,6 +263,15 @@ class UIKitDemoViewController: UIViewController {
         return true
     }
 
+    @objc func onSettingsPressed() {
+        let settingsView = SettingsView(envConfigSubject: envConfigSubject, dismissAction: {
+            self.dismiss(animated: true, completion: nil)
+        })
+        let settingsViewHostingController = UIHostingController(rootView: settingsView)
+        settingsViewHostingController.modalPresentationStyle = .fullScreen
```
Comment (kevinyulu) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Views/UIKitDemoViewController.swift:271]: nice to have: on iPad this is covering the whole screen, in the future we may want to use other presentation style.

**Review by vhuseinova-msft — APPROVED** [review]
I like  the Jim's idea about mooving the Settings button. Otherwise, LGTM, tested.

**Review by kevinyulu — APPROVED** [review]
tested on iPhone XR iOS 15.3, 
Settings button displays above Start Experience button, language settings works as expected. 
Tested the UX in portrait and landscape. UX looks good. 
Also tested light and dark mode. looks good.

**Comment by JoshuaLai** [comment]
> I like the Jim's idea about mooving the Settings button. Otherwise, LGTM, tested.

Agreed. Jim and I chatted about it. To do that I think best practice would be for our entry view to implement a navigation controller. And use an accessory in that. It would be a larger refactor so we will track it as a upcoming refactor task

**Pull request merged**